motto/agents/metta_agent.py

```diff
@@ -62,6 +62,10 @@ def __call__(self, msgs_atom, functions=[]):
             #response = metta.load_module_at_path(self._path)
             with open(self._path, mode='r') as f:
                 code = f.read()
+                # what to do if need to set som variables from python?
+                if (variables_values is not None) and isinstance(variables_values, dict):
```

Comment:
In the awakening-simple-chitchat system, a system message containing `situation_awareness` and current time is added to the messages. This system message is not saved to the dialogue history. To achieve the same behavior with metta-motto, I made the following call:

  ```(llm (Agent (chat-gpt-ext "gpt-3.5-turbo" False True)) (Messages (history) (system $system) (messages)))```
  The variable `$system` needs to be passed into the agent, so I added the `variables_values` parameter for `MettaAgent`. This might not be the best solution. @Necr0x0Der, what do you suggest?

Reply:
The name of the model can also be set by the same way.

Reply:
Thanks. Indeed, we need to have a way to pass additional info to metta agents. However, I'm not sure that replacing variables in the code is the best way to achieve this. I'd suggest to consider adding functions (similar to `history` or `messages`) for this purpose. Say, instead of having `$system` to be replaced with `"my system message"`, we can have `(system_msg)` and add `(= (system_msg) "my system message")`. This mechanism can be later extended to prepend the agent with any additional code.